File: enum_switch.c
#include <stdio.h>
#include <stdlib.h>

enum Shape {
    Circle,
    Triangle,
    Square
};

void print_shape(enum Shape shape) {
    switch (shape) {
        case Circle:    printf("shape is circle\n");      break;
        case Triangle:  printf("shape is triangle\n");    break;
        case Square:    printf("shape is square\n");      break;
    }
}

int main() 
{
    print_shape(Circle);
    print_shape(Triangle);
    print_shape(Square);
} 

File: enum_switch.rs
enum Shape {
    Circle,
    Triangle,
    Square
}

fn print_shape(shape: Shape) {
    match shape {
        Shape::Circle => println!("shape is circle"),
        Shape::Triangle => println!("shape is triangle"),
        Shape::Square => println!("shape is square"),
    }
}

fn main() {
    print_shape(Shape::Circle);
    print_shape(Shape::Triangle);
    print_shape(Shape::Square);
}